scripts/build-docs.ts

```diff
@@ -0,0 +1,126 @@
+
+const docsConfig: DocsConfig = {
+    mode: "file",
+    name: "Client App SDK",
+    logger: 'none',
+    theme: "docusaurus", // Theme that generates output closest to Dev Center format
+    skipSidebar: true, // Disable sidebar generation
+    disableSources: true, // Don't show the file for each method/class
+    hideBreadcrumbs: true, // Dev Center handles breadcrumbs for us
+    readme: "none",
+    plugin: [
+        'typedoc-plugin-markdown',
+        'typedoc-plugin-no-inherit'
+    ],
+    exclude: [
+        "**/utils/**",
+        "**/.babelrc",
+        "**/.eslintrc.js",
+        "**/*+(Spec).ts"
+    ],
+    excludeExternals: true,
+    excludePrivate: true,
+    excludeProtected: true,
+    excludeNotExported: true,
+    excludeNotDocumented: true
+};
+
+const transformLinks = (buffer: string, ext: string) => {
+    // Regex to replace the following patterns (ext = "html"):
+    // [link1](api.md) -> [link1](api.html)
+    // [link2](api.md#someref) -> [link2](api.html#someref)
+    return buffer.replace(/(\[[^\]]+\][^)]+)(\.md)(\)|#[^)]*\))/gm, `$1${ext}$3`);
+};
+
+const prependDocHeaderAttribute = (buffer: string, attr: string) => {
+    if (!buffer.startsWith('---')) {
+        throw new Error('Doc header symbol "---" not found');
+    }
+    return buffer = `---\n${attr}${buffer.substring('---'.length)}`;
+};
+
+(async () => {
+    const app = new Application();
+
+    // So TypeDoc can load tsconfig.json
+    app.options.addReader(new TSConfigReader());
+
+    app.bootstrap(docsConfig);
+
+    const project = app.convert(app.expandInputFiles(["src"]));
+    if (!project) throw new Error("Unable to generate typedoc project");
+
+    // Rendered docs
+    app.generateDocs(project, OUTPUT_DIR);
+
+    // Flatten typedoc's generated classes folder
+    const classesDir = path.join(OUTPUT_DIR, "classes");
+    const fileNames = await fs.readdir(classesDir);
+    await Promise.all(
+        fileNames.map(async (name) =>
+            fs.rename(path.join(classesDir, name), path.join(OUTPUT_DIR, name))
+        )
+    );
+
+    // Delete typedoc's generated classes folder
+    await fs.remove(classesDir);
+
+    // Use custom index file in place of typedoc's index
+    const indexFilePath = path.join(OUTPUT_DIR, "index.md");
+    await fs.copy(INDEX_FILE, indexFilePath);
+
+    // Add header to index file
+    const indexFileContents = (await fs.readFile(indexFilePath)).toString();
+    await fs.outputFile(indexFilePath, "---\ntitle: Client App SDK\n---\n\n" + indexFileContents);
+
+    // Apply transformations to docs
+    const docs = await fs.readdir(OUTPUT_DIR);
+    await Promise.all(
+        docs.map(async (doc) => {
+            const docPath = path.join(OUTPUT_DIR, doc);
+            let buffer = (await fs.readFile(docPath)).toString();
+            if (docMdOutputFormat === PC_DEV_CENTER_FORMAT) {
+                buffer = transformLinks(buffer, '.html');
```

Comment:
Link transform is looking good.  Working between modules and with anchor tags.  The only issue I've seen when running locally is the ./ClientApp.md links in index.html.  Those links seem to be generated with a target="_blank" and open in new tabs.

Not sure if this is an issue with the external link detection algo in dev center or if it's just from running locally.  Maybe @PrinceNebulon could weigh in on this one as well.

Reply:
Can you provide a specific example? Is there a link on the dev center that's not working correctly currently?